jupyter_server/auth/identity.py

```diff
@@ -0,0 +1,155 @@
+"""Identity Provider interface
+
+This defines the _authentication_ layer of Jupyter Server,
+to be used in combination with Authorizer for _authorization_.
+
+.. versionadded:: 2.0
+"""
+import sys
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Optional
+
+from tornado.web import RequestHandler
+from traitlets.config import LoggingConfigurable
+
+if sys.version_info >= (3, 8):
+    from typing import TypedDict
+else:
+    try:
+        from typing_extensions import TypedDict
+    except ImportError:
+        TypedDict = Dict
+
+
+class IdentityModel(TypedDict):
+    # see the JupyterLab IUser model for definitions
+
+    username: str  # the only truly required field
+
+    # these fields are derived from username if not specified
+    name: str
+    display_name: str
```

Comment:
I confess I, too, don't know when these would be different or what they are for. They are copied from JupyterLab. Maybe it's for something like 'real name' and 'preferred/nick name'? e.g. `name: Benjamin Ragan-Kelley, display_name: Min RK` for me?

Reply:
That makes sense and I suppose that's an application decision as to which to use (and when).  Would it be helpful to provide a _usage hint_ for each, or might that be overstepping things?  Just trying to prevent a situation where one "application" views `name` vs. `display_name` differently from another, and perhaps providing loosely-worded _hints_ might help avoid that situation (assuming boh are retained).